### PR TITLE
Add `fillet_wire_2D` and `chamfer_wire_2D`

### PR DESCRIPTION
**Pull request by CoronelBuendia**
## Linked Issues

Closes #{ID}

## Checklist

I confirm that I have completed the following checks:

- [x] Tests run locally and pass `pytest tests --reactor`
- [x] Code quality checks run locally and pass `flake8` and `black .`
- [x] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: bluemira/codes/_freecadapi.py [hunk 1]
```diff
@@ -1324,6 +1326,31 @@ def sweep_shape(profiles, path, solid=True, frenet=True):
         return solid_result.Shells[0]
 
 
+def fillet_wire_2D(wire: apiWire, radius: float, chamfer: bool = False) -> apiWire:
+    """
```
Comment (ivanmaione) [inline bluemira/codes/_freecadapi.py:1330]: Is it necessary to check that the wire is planar or is it up to the user?

File: tests/geometry/test_tools.py [hunk 2]
```diff
@@ -712,3 +715,44 @@ def test_awkward_mirror(self, shape):
     def test_bad_direction(self, shape):
         with pytest.raises(GeometryError):
             mirror_shape(shape, base=(0, 0, 0), direction=(EPS, EPS, EPS))
+
+
+class TestFilletChamfer2D:
+    closed_rectangle = make_polygon({"x": [0, 2, 2, 0], "z": [0, 0, 2, 2]}, closed=True)
+    open_rectangle = make_polygon({"x": [0, 2, 2, 0], "z": [0, 0, 2, 2]}, closed=False)
+
```
Comment (ivanmaione) [inline tests/geometry/test_tools.py:723]: missing a test for negative radius

File: bluemira/geometry/tools.py [hunk 3]
```diff
@@ -758,6 +758,54 @@ def sweep_shape(profiles, path, solid=True, frenet=True, label=""):
...
+    radius:
+        Radius of the fillet operation
+
+    Returns
+    -------
+    filleted_wire:
+        The filleted wire
+    """
+    if radius == 0:
+        return wire.deepcopy()
```
Comment (ivanmaione) [inline bluemira/geometry/tools.py:778]: I would have raised an error here since the operation cannot be performed. However, if there are particular applications in which this behaviour can be useful, I have nothing in particular against this solution.

File: bluemira/codes/_freecadapi.py (hunk 1 above)
Comment (CoronelBuendia) [reply bluemira/codes/_freecadapi.py:1330]: It does actually work for 3-D wires, but I dare not expose that just yet... I will put in a defense against non-planar wires

File: bluemira/geometry/tools.py (hunk 3 above)
Comment (CoronelBuendia) [reply bluemira/geometry/tools.py:778]: Yeah that's fair. It seems we've tried to be "forgiving" in most places where we can be. So an offset of 0 also gives a deepcopy. FreeCAD also gives you the same object in this case (but copied), so I just went with the flow

File: bluemira/codes/_freecadapi.py [hunk 4]
```diff
@@ -1324,6 +1328,44 @@ def sweep_shape(profiles, path, solid=True, frenet=True):
...
+    if not _wire_is_planar(wire):
+        raise GeometryError(
+            f"Cannot {'chamfer' if chamfer else 'fillet'} a non-planar wire!"
+        )
+
+    # Temporarily suppress pesky print statement:
+    # DraftGeomUtils.fillet: Warning: edges have same direction. Did nothing
+    sys.stdout = open(os.devnull, "w")
+    result = DraftGeomUtils.filletWire(wire, radius, chamfer=chamfer)
+    sys.stdout = sys.__stdout__
```
[the remark this answers is not in the document]
Comment (je-cook) [reply bluemira/codes/_freecadapi.py:1364]: I tried suppressing it with warnings (see https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings) but its apparently not using pythons mechanics so not really.
 
I dont think it should change anything for us, we dont muck with stdout logging just uses the default, especially as you put it back immediately after

[the remark this answers is not in the document]
Comment (CoronelBuendia) [reply bluemira/codes/_freecadapi.py:1364]: Nah, it's just a print statement straight from Python in FreeCAD. OK, so all good on the logging front then!

[the remark this answers is not in the document]
Comment (hsaunders1904) [reply bluemira/codes/_freecadapi.py:1364]: I would wrap this in a try-finally block, though.

```suggestion
    # Temporarily suppress pesky print statement:
    # DraftGeomUtils.fillet: Warning: edges have same direction. Did nothing
    old_stdout = sys.stdout
    try:
        sys.stdout = open(os.devnull, "w")
        result = DraftGeomUtils.filletWire(wire, radius, chamfer=chamfer)
    finally:
        sys.stdout = old_stdout
```

File: bluemira/geometry/tools.py [hunk 5]
```diff
@@ -758,6 +758,73 @@ def sweep_shape(profiles, path, solid=True, frenet=True, label=""):
...
 
 
+def fillet_chamfer_decorator(chamfer: bool):
+    """
+    Decorator for fillet and chamger operations, checking for validity of wire
+    and radius.
+    """
+
+    def decorator(func):
+        def wrapper(wire, radius):
```
Comment (je-cook) [inline bluemira/geometry/tools.py:768]: We get all the typing and return types properly and the doctrings get copied over as well if we do this

```suggestion
        @functools.wraps(func)
        def wrapper(wire, radius):
```

**Review by je-cook — APPROVED** [review]

**Review by ivanmaione — APPROVED** [review]
